### ISSUE TITLE
Missing keys during training

### ISSUE THREAD

**Issue by PaulSudarshan** [SuperMHP/GUPNet_Plus#9]
Hi,
I was trying to train the model on nuscenes dataset, while launching training I came across the following log, however the training works fine, just want to make sure everything is right, and is this expected?

unexpected key in source state_dict: fc.weight, fc.bias, level2.root.bn.weight, level2.root.bn.bias, level2.root.bn.running_mean, level2.root.bn.running_var, level2.tree1.bn1.weight, level2.tree1.bn1.bias, level2.tree1.bn1.running_mean, level2.tree1.bn1.running_var, level2.tree1.bn2.weight, level2.tree1.bn2.bias, level2.tree1.bn2.running_mean, level2.tree1.bn2.running_var, level2.tree2.bn1.weight, level2.tree2.bn1.bias, level2.tree2.bn1.running_mean, level2.tree2.bn1.running_var, level2.tree2.bn2.weight, level2.tree2.bn2.bias, level2.tree2.bn2.running_mean, level2.tree2.bn2.running_var, level3.tree1.root.bn.weight, level3.tree1.root.bn.bias, level3.tree1.root.bn.running_mean, level3.tree1.root.bn.running_var, level3.tree1.tree1.bn1.weight, level3.tree1.tree1.bn1.bias, level3.tree1.tree1.bn1.running_mean, level3.tree1.tree1.bn1.running_var, level3.tree1.tree1.bn2.weight, level3.tree1.tree1.bn2.bias, level3.tree1.tree1.bn2.running_mean, level3.tree1.tree1.bn2.running_var, level3.tree1.tree2.bn1.weight, level3.tree1.tree2.bn1.bias, level3.tree1.tree2.bn1.running_mean, level3.tree1.tree2.bn1.running_var, level3.tree1.tree2.bn2.weight, level3.tree1.tree2.bn2.bias, level3.tree1.tree2.bn2.running_mean, level3.tree1.tree2.bn2.running_var, level3.tree2.root.bn.weight, level3.tree2.root.bn.bias, level3.tree2.root.bn.running_mean, level3.tree2.root.bn.running_var, level3.tree2.tree1.bn1.weight, level3.tree2.tree1.bn1.bias, level3.tree2.tree1.bn1.running_mean, level3.tree2.tree1.bn1.running_var, level3.tree2.tree1.bn2.weight, level3.tree2.tree1.bn2.bias, level3.tree2.tree1.bn2.running_mean, level3.tree2.tree1.bn2.running_var, level3.tree2.tree2.bn1.weight, level3.tree2.tree2.bn1.bias, level3.tree2.tree2.bn1.running_mean, level3.tree2.tree2.bn1.running_var, level3.tree2.tree2.bn2.weight, level3.tree2.tree2.bn2.bias, level3.tree2.tree2.bn2.running_mean, level3.tree2.tree2.bn2.running_var, level4.tree1.root.bn.weight, level4.tree1.root.bn.bias, level4.tree1.root.bn.running_mean, level4.tree1.root.bn.running_var, level4.tree1.tree1.bn1.weight, level4.tree1.tree1.bn1.bias, level4.tree1.tree1.bn1.running_mean, level4.tree1.tree1.bn1.running_var, level4.tree1.tree1.bn2.weight, level4.tree1.tree1.bn2.bias, level4.tree1.tree1.bn2.running_mean, level4.tree1.tree1.bn2.running_var, level4.tree1.tree2.bn1.weight, level4.tree1.tree2.bn1.bias, level4.tree1.tree2.bn1.running_mean, level4.tree1.tree2.bn1.running_var, level4.tree1.tree2.bn2.weight, level4.tree1.tree2.bn2.bias, level4.tree1.tree2.bn2.running_mean, level4.tree1.tree2.bn2.running_var, level4.tree2.root.bn.weight, level4.tree2.root.bn.bias, level4.tree2.root.bn.running_mean, level4.tree2.root.bn.running_var, level4.tree2.tree1.bn1.weight, level4.tree2.tree1.bn1.bias, level4.tree2.tree1.bn1.running_mean, level4.tree2.tree1.bn1.running_var, level4.tree2.tree1.bn2.weight, level4.tree2.tree1.bn2.bias, level4.tree2.tree1.bn2.running_mean, level4.tree2.tree1.bn2.running_var, level4.tree2.tree2.bn1.weight, level4.tree2.tree2.bn1.bias, level4.tree2.tree2.bn1.running_mean, level4.tree2.tree2.bn1.running_var, level4.tree2.tree2.bn2.weight, level4.tree2.tree2.bn2.bias, level4.tree2.tree2.bn2.running_mean, level4.tree2.tree2.bn2.running_var, level5.root.bn.weight, level5.root.bn.bias, level5.root.bn.running_mean, level5.root.bn.running_var, level5.tree1.bn1.weight, level5.tree1.bn1.bias, level5.tree1.bn1.running_mean, level5.tree1.bn1.running_var, level5.tree1.bn2.weight, level5.tree1.bn2.bias, level5.tree1.bn2.running_mean, level5.tree1.bn2.running_var, level5.tree2.bn1.weight, level5.tree2.bn1.bias, level5.tree2.bn1.running_mean, level5.tree2.bn1.running_var, level5.tree2.bn2.weight, level5.tree2.bn2.bias, level5.tree2.bn2.running_mean, level5.tree2.bn2.running_var

missing keys in source state_dict: level2.root.norm.weight, level2.root.norm.bias, level2.root.norm.running_mean, level2.root.norm.running_var, level2.tree1.norm1.weight, level2.tree1.norm1.bias, level2.tree1.norm1.running_mean, level2.tree1.norm1.running_var, level2.tree1.norm2.weight, level2.tree1.norm2.bias, level2.tree1.norm2.running_mean, level2.tree1.norm2.running_var, level2.tree2.norm1.weight, level2.tree2.norm1.bias, level2.tree2.norm1.running_mean, level2.tree2.norm1.running_var, level2.tree2.norm2.weight, level2.tree2.norm2.bias, level2.tree2.norm2.running_mean, level2.tree2.norm2.running_var, level3.tree1.root.norm.weight, level3.tree1.root.norm.bias, level3.tree1.root.norm.running_mean, level3.tree1.root.norm.running_var, level3.tree1.tree1.norm1.weight, level3.tree1.tree1.norm1.bias, level3.tree1.tree1.norm1.running_mean, level3.tree1.tree1.norm1.running_var, level3.tree1.tree1.norm2.weight, level3.tree1.tree1.norm2.bias, level3.tree1.tree1.norm2.running_mean, level3.tree1.tree1.norm2.running_var, level3.tree1.tree2.norm1.weight, level3.tree1.tree2.norm1.bias, level3.tree1.tree2.norm1.running_mean, level3.tree1.tree2.norm1.running_var, level3.tree1.tree2.norm2.weight, level3.tree1.tree2.norm2.bias, level3.tree1.tree2.norm2.running_mean, level3.tree1.tree2.norm2.running_var, level3.tree2.root.norm.weight, level3.tree2.root.norm.bias, level3.tree2.root.norm.running_mean, level3.tree2.root.norm.running_var, level3.tree2.tree1.norm1.weight, level3.tree2.tree1.norm1.bias, level3.tree2.tree1.norm1.running_mean, level3.tree2.tree1.norm1.running_var, level3.tree2.tree1.norm2.weight, level3.tree2.tree1.norm2.bias, level3.tree2.tree1.norm2.running_mean, level3.tree2.tree1.norm2.running_var, level3.tree2.tree2.norm1.weight, level3.tree2.tree2.norm1.bias, level3.tree2.tree2.norm1.running_mean, level3.tree2.tree2.norm1.running_var, level3.tree2.tree2.norm2.weight, level3.tree2.tree2.norm2.bias, level3.tree2.tree2.norm2.running_mean, level3.tree2.tree2.norm2.running_var, level4.tree1.root.norm.weight, level4.tree1.root.norm.bias, level4.tree1.root.norm.running_mean, level4.tree1.root.norm.running_var, level4.tree1.tree1.norm1.weight, level4.tree1.tree1.norm1.bias, level4.tree1.tree1.norm1.running_mean, level4.tree1.tree1.norm1.running_var, level4.tree1.tree1.norm2.weight, level4.tree1.tree1.norm2.bias, level4.tree1.tree1.norm2.running_mean, level4.tree1.tree1.norm2.running_var, level4.tree1.tree2.norm1.weight, level4.tree1.tree2.norm1.bias, level4.tree1.tree2.norm1.running_mean, level4.tree1.tree2.norm1.running_var, level4.tree1.tree2.norm2.weight, level4.tree1.tree2.norm2.bias, level4.tree1.tree2.norm2.running_mean, level4.tree1.tree2.norm2.running_var, level4.tree2.root.norm.weight, level4.tree2.root.norm.bias, level4.tree2.root.norm.running_mean, level4.tree2.root.norm.running_var, level4.tree2.tree1.norm1.weight, level4.tree2.tree1.norm1.bias, level4.tree2.tree1.norm1.running_mean, level4.tree2.tree1.norm1.running_var, level4.tree2.tree1.norm2.weight, level4.tree2.tree1.norm2.bias, level4.tree2.tree1.norm2.running_mean, level4.tree2.tree1.norm2.running_var, level4.tree2.tree2.norm1.weight, level4.tree2.tree2.norm1.bias, level4.tree2.tree2.norm1.running_mean, level4.tree2.tree2.norm1.running_var, level4.tree2.tree2.norm2.weight, level4.tree2.tree2.norm2.bias, level4.tree2.tree2.norm2.running_mean, level4.tree2.tree2.norm2.running_var, level5.root.norm.weight, level5.root.norm.bias, level5.root.norm.running_mean, level5.root.norm.running_var, level5.tree1.norm1.weight, level5.tree1.norm1.bias, level5.tree1.norm1.running_mean, level5.tree1.norm1.running_var, level5.tree1.norm2.weight, level5.tree1.norm2.bias, level5.tree1.norm2.running_mean, level5.tree1.norm2.running_var, level5.tree2.norm1.weight, level5.tree2.norm1.bias, level5.tree2.norm1.running_mean, level5.tree2.norm1.running_var, level5.tree2.norm2.weight, level5.tree2.norm2.bias, level5.tree2.norm2.running_mean, level5.tree2.norm2.running_var

**Comment by SuperMHP** [comment]
It is ok. My experiments also showed these warnings. I have also tried before. Current setting (without pre-trained BN) works well on small datasets, e. g. KITTI.

**Comment by SuperMHP** [comment]
samples_per_gpu=16 and num_gpus=2 （bsize=32）are not consistent with our recommended settings on nuScenes. In our original paper, we stated that the batch size should be set to 128, the same as in CenterTrack. From our experiments, using a smaller batch size leads to a large performance drop and is hard to converge, mainly because the nuScenes dataset is challenging and the gradients can be quite noisy. If you prefer to train with a smaller batch size, you will need to carefully tune the hyperparameters. However, we have not yet found a configuration that works well under such conditions.

**Comment by SuperMHP** [comment]
Please check the README Test section
